ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -181,16 +181,23 @@ public EndpointResponse streamQuery(
       final Optional<Boolean> isInternalRequest,
       final KsqlMediaType mediaType
   ) {
-    throwIfNotConfigured();
-    activenessRegistrar.updateLastRequestTime();
+    EndpointResponse response = null;
+    try {
+      throwIfNotConfigured();
+      activenessRegistrar.updateLastRequestTime();
 
-    final PreparedStatement<?> statement = parseStatement(request);
+      final PreparedStatement<?> statement = parseStatement(request);
 
-    CommandStoreUtil.httpWaitForCommandSequenceNumber(
-        commandQueue, request, commandQueueCatchupTimeout);
+      CommandStoreUtil.httpWaitForCommandSequenceNumber(
```

Comment:
nit: we should remove this file from the PR (sorry leftover from my proof of concept)